ui/src/app/applications/components/search-bar.tsx

```diff
@@ -0,0 +1,96 @@
+    const {content, ctx, apps} = {...props};
+
+    const searchBar = React.useRef<HTMLDivElement>(null);
+
+    const query = new URLSearchParams(window.location.search);
+    const appInput = tryJsonParse(query.get('new'));
+
+    const {useKeybinding} = React.useContext(KeybindingContext);
+    const [isFocused, setFocus] = React.useState(false);
+
+    useKeybinding({
+        keys: Key.SLASH,
+        action: () => {
+            if (searchBar.current && !appInput) {
+                searchBar.current.querySelector('input').focus();
+                setFocus(true);
+                return true;
+            }
+            return false;
+        }
+    });
+
+    useKeybinding({
+        keys: Key.ESCAPE,
+        action: () => {
+            if (searchBar.current && !appInput && isFocused) {
+                searchBar.current.querySelector('input').blur();
+                setFocus(false);
+                return true;
+            }
+            return false;
+        }
+    });
+
+    return (
+        <Autocomplete
+            filterSuggestions={true}
+            renderInput={inputProps => (
+                <div className='applications-list__search' ref={searchBar}>
+                    <i
+                        className='fa fa-search'
+                        style={{marginRight: '9px', cursor: 'pointer'}}
+                        onClick={() => {
+                            if (searchBar.current) {
+                                searchBar.current.querySelector('input').focus();
+                            }
+                        }}
+                    />
+                    <input
+                        {...inputProps}
+                        onFocus={e => {
+                            e.target.select();
+                            if (inputProps.onFocus) {
+                                inputProps.onFocus(e);
+                            }
+                        }}
+                        style={{fontSize: '14px'}}
+                        className='argo-field'
+                        placeholder='Search applications...'
+                    />
+                    <div className='keyboard-hint'>/</div>
+                    {content && (
+                        <i className='fa fa-times' onClick={() => ctx.navigation.goto('.', {search: null}, {replace: true})} style={{cursor: 'pointer', marginLeft: '5px'}} />
+                    )}
+                </div>
+            )}
+            wrapperProps={{className: 'applications-list__search-wrapper'}}
+            renderItem={item => (
+                <React.Fragment>
+                    <i className='icon argo-icon-application' /> {item.label}
+                </React.Fragment>
+            )}
+            onSelect={val => {
+                ctx.navigation.goto(`/applications/${val}`);
+            }}
+            onChange={e => ctx.navigation.goto('.', {search: e.target.value}, {replace: true})}
```

Comment:
If we are already in the details page for that application, then should we 'go' there again?

Reply:
If you mean the `onChange` part, it's from the original implementation. I think only in this way it will go to the searched app page. Because different app page has different URL and it should be pushed into navigation history, correct me if I was wrong.

Reply:
Yes, in the original implementation, it _has_ to go to the details page because it will be a different URL.  However, I'm referring to the case where you are already in the details page for that app, and then you try to select the same app again (since it is in the search bar list).  I think you have to account for that.

Reply:
As discussed, we want to gray out the current application from the search autocomplete list, so that it won't add extra history.

Reply:
To make the item disabled on the autocomplete list I have to update the argo-ui repo. Instead I made the label for current application display as `some-app (current app)`, also added a condition on the navigation, so that it doesn't add navigation history when current app gets clicked. Let me know if this is acceptable. Thanks!

<img width="331" alt="Screen Shot 2021-12-06 at 6 49 34 PM" src="https://user-images.githubusercontent.com/26255262/144940950-d55876d5-9623-46be-b2e3-8eadc2a454a5.png">

Reply:
@ciiay, ok, if needed, we can change this later, but what you have here works fine.